src/lib/validate-declarative-schema.js

```diff
@@ -104,8 +104,12 @@ const TaskSchema = joi.array().items(
     contactLabel:
       joi.alternatives().try( joi.string().min(1), joi.function() ).optional()
       .error(taskError('"contactLabel" should either be a non-empty string or function(contact, report)')),
-    resolvedIf: joi.function().required()
-      .error(taskError('"resolvedIf" should be of type function(contact, report)')),
+    resolvedIf: joi.alternatives().conditional('actions', {
+      is: joi.array().items(joi.object({ type: 'contact' }).unknown()),
```

Comment:
Thanks for this perspective. Did you mean something like this?
```js
    resolvedIf: joi
      .alternatives()
      .conditional('actions', {
        is: joi.array().has(joi.object({ type: 'report' }).unknown()),
        then: joi.function().optional(),
        otherwise: joi.function().required()
      })
```

Reply:
Looks good! :)